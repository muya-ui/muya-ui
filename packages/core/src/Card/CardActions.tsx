import React from 'react';
import { ICardActionsProps } from './types';
import { StyledCardActions } from './styled';
import useTheme from '../utils/useTheme';
import memoForwardRef from '../utils/memoForwardRef';

const CardActions = memoForwardRef<HTMLDivElement, ICardActionsProps>((props, ref) => {
  const { children, padding, bordered = true, ...others } = props;
  const theme = useTheme();

  return (
    <StyledCardActions theme={theme} $padding={padding} $bordered={bordered} ref={ref} {...others}>
      {children}
    </StyledCardActions>
  );
});

export default CardActions;
