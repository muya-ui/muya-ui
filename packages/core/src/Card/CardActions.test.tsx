import React from 'react';
import renderer from 'react-test-renderer';
import CardActions from './CardActions';

describe('CardActions', () => {
  it('should render success', function() {
    const wrapper = renderer
      .create(
        <>
          <CardActions>测试</CardActions>
          <CardActions padding={0}>测试</CardActions>
          <CardActions bordered={false}>测试</CardActions>
        </>,
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
