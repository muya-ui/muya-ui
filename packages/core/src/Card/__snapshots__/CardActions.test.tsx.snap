// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardActions should render success 1`] = `
Array [
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 16px 20px;
  border-top: 1px solid rgba(238,238,238,1);
}

<div
    className="c0"
  >
    测试
  </div>,
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  border-top: 1px solid rgba(238,238,238,1);
}

<div
    className="c0"
  >
    测试
  </div>,
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 16px 20px;
}

<div
    className="c0"
  >
    测试
  </div>,
]
`;
